lib/NOM/Print.hs

```diff
@@ -151,8 +153,8 @@ printErrors errors maxHeight =
 compactError :: Text -> Text
 compactError = fst . Text.breakOn "\n       last 10 log lines:"
 
-stateToText :: Config -> NOMV1State -> Maybe (Window Int) -> (ZonedTime, Double) -> Text
-stateToText config buildState@MkNOMV1State{..} = memo printWithSize . fmap Window.height
+stateToText :: Config -> NOMV1State -> PrintState -> Maybe (Window Int) -> (ZonedTime, Double) -> Text
+stateToText config buildState@MkNOMV1State{..} printState = memo printWithSize . fmap Window.height
```

Comment:
This might be a problem. What I was trying to do here is to memoize this whole function. It should only be called new when the state or the window size changes. The function tries to precalculate most of its state and only use the input time event a little. (This might be overengineered and I am not sure whether I ever benchmarked this.)

Never-the-less passing in the new print state on every print event will invalidate the memoisation here.

Reply:
Right, I see. Hmm. I'm not familiar with memotrie, I'll have to read a bit about it to understand how it works.